data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -219,7 +230,9 @@ private List<DeleteMessageBatchRequestEntry> processS3EventNotificationRecords(f
             AcknowledgementSet acknowledgementSet = null;
             if (endToEndAcknowledgementsEnabled) {
                 // Acknowledgement Set timeout is slightly smaller than the visibility timeout;
-                int timeout = (int) sqsOptions.getVisibilityTimeout().getSeconds() - 2;
+                int timeout = Objects.isNull(sqsDynamicVisibilityTimeoutManager) ?
+                        (int) sqsOptions.getVisibilityTimeout().getSeconds() - 2 :
+                        sqsDynamicVisibilityTimeoutManager.getRemainingTimeBeforeNextChangeMessageIncrease(parsedMessage.message.messageId());
```

Comment:
I think this will always be the initial timeout since the startDynamic call doesn't occur until after this. I'm not sure what the best way to keep the E2E acks in sync would be

Reply:
@kkondaka , See above.

Reply:
I think we need a new API to AcknowledgementSet to change the timeout value. I think that's the right thing do. So, at the time of creation of  it would be`sqsOptions.getVisibilityTimeout().getSeconds() - 2`  (whether visibility timeout is static or dynamic. But when the new value for visibility time is determined, we need to make a call to AcknowledgementSet to change it's timeout value - something like - `acknowledgementSet.changeTimeout(newValue)`